packages/svelte/src/internal/client/runtime.js

```diff
@@ -157,77 +154,44 @@ export function is_runes() {
  */
 export function check_dirtiness(reaction) {
 	var flags = reaction.f;
-	var is_dirty = (flags & DIRTY) !== 0;
 
-	if (is_dirty) {
+	if ((flags & DIRTY) !== 0) {
 		return true;
 	}
 
-	var is_unowned = (flags & UNOWNED) !== 0;
-	var is_disconnected = (flags & DISCONNECTED) !== 0;
-
 	if ((flags & MAYBE_DIRTY) !== 0) {
 		var dependencies = reaction.deps;
 
 		if (dependencies !== null) {
-			var length = dependencies.length;
-			var reactions;
+			var is_unowned = (flags & UNOWNED) !== 0;
 
-			for (var i = 0; i < length; i++) {
+			for (var i = 0; i < dependencies.length; i++) {
 				var dependency = dependencies[i];
 
-				if (!is_dirty && check_dirtiness(/** @type {import('#client').Derived} */ (dependency))) {
-					update_derived(/** @type {import('#client').Derived} **/ (dependency));
+				if (check_dirtiness(/** @type {import('#client').Derived} */ (dependency))) {
+					update_derived(/** @type {import('#client').Derived} */ (dependency));
 				}
 
-				if ((reaction.f & DIRTY) !== 0) {
-					// `reaction` might now be dirty, as a result of calling `update_derived`
+				if (dependency.version > reaction.version) {
 					return true;
 				}
 
 				if (is_unowned) {
-					// If we're working with an unowned derived signal, then we need to check
-					// if our dependency write version is higher. If it is then we can assume
-					// that state has changed to a newer version and thus this unowned signal
-					// is also dirty.
-					if (dependency.version > /** @type {import('#client').Derived} */ (reaction).version) {
-						return true;
-					}
-
+					// TODO is there a more logical place to do this work?
 					if (!current_skip_reaction && !dependency?.reactions?.includes(reaction)) {
 						// If we are working with an unowned signal as part of an effect (due to !current_skip_reaction)
 						// and the version hasn't changed, we still need to check that this reaction
 						// if linked to the dependency source – otherwise future updates will not be caught.
 						(dependency.reactions ??= []).push(reaction);
 					}
-				} else if (is_disconnected) {
-					// It might be that the derived was was dereferenced from its dependencies but has now come alive again.
-					// In thise case, we need to re-attach it to the graph and mark it dirty if any of its dependencies have
-					// changed since.
-					if (dependency.version > /** @type {import('#client').Derived} */ (reaction).version) {
-						is_dirty = true;
-					}
-
-					reactions = dependency.reactions;
-					if (reactions === null) {
-						dependency.reactions = [reaction];
-					} else if (!reactions.includes(reaction)) {
-						reactions.push(reaction);
-					}
 				}
 			}
 		}
 
-		// Unowned signals are always maybe dirty, as we instead check their dependency versions.
-		if (!is_unowned) {
-			set_signal_status(reaction, CLEAN);
-		}
-		if (is_disconnected) {
-			reaction.f ^= DISCONNECTED;
-		}
+		set_signal_status(reaction, CLEAN);
```

Comment:
So historically, unowned signals are generally always kept as maybe_dirty, that's so we can check their version. If they are unowned and they're not within an effect, nothing will mark them as maybe dirty again, so they basically stay clean forever. I wonder if you've avoided that somehow though? I guess we can always come back to this if not.

Reply:
I added a test, and it looks good :)

Reply:
Turns out that we needed this still: https://github.com/sveltejs/svelte/pull/12747